source/pi/warble.sh

```diff
@@ -5,9 +5,43 @@ CODE=$2
 URL=$3
 TWEET=$4
 
+function check_environment {
+  if [ -z "$JAVA_HOME" ] || [ -z "$GRAALVISOR_PATH" ]
+  then
+    echo "Please check your environment variables."
+    exit 1
+  fi
+}
+
+function start_graalvisor {
+  export lambda_timestamp="$(date +%s%N | cut -b1-13)"
+  export lambda_port="$GRAALVISOR_PORT"
+  $GRAALVISOR_PATH
+}
+
+function register_function {
+  curl -s -X POST $GRAALVISOR_IP:$GRAALVISOR_PORT/register?name=warble\&entryPoint=main\&language=python -H 'Content-Type: application/json' --data-binary @main.py
+}
+
 pushd ../warble
 
-OUTPUT=$(python3 warblecc.py --username ${USERNAME} ${CODE})
+check_environment
+GRAALVISOR_IP=127.0.0.1
+GRAALVISOR_PORT=8080
+# Checking if the default port of GraalVisor is in use.
+if [ -z "$(sudo lsof -i -P -n | grep LISTEN | grep $GRAALVISOR_PORT)" ]
+then
+  # GraalVisor is down, we have to launch it and register the function.
+  start_graalvisor &> "$GRAALVISOR_PATH".log &
```

Comment:
If graalvisor is on a shared network mount (NSF), this will not be a good idea. We should do /tmp/graalvisor.log.